src/gui/editorwidgets/qgsdatetimeedit.h

```diff
@@ -38,6 +38,20 @@ class GUI_EXPORT QgsDateTimeEdit : public QDateTimeEdit
     void setAllowNull( bool allowNull );
     bool allowNull() const {return mAllowNull;}
 
+    /** Set the lowest Date that can be displayed with the Qt::ISODate format
+     *  - uses QDateTimeEdit::setMinimumDateTime (since Qt 4.4)
+     * \note
+     *  - QDate and QDateTime does not support minus years for the Qt::ISODate format
+     *  -> returns empty (toString) or invalid (fromString) values
+     *  - QDateTimeEdit::setMinimumDateTime does not support dates < '0100-01-01'
+     *  -> it is not for us to wonder why [defined in qdatetimeparser_p.h]
+    * \since QGIS 3.0
+    */
+    void setMinimumEditDateTime()
```

Comment:
You need to make the method private in the .h , then rerun scripts/prepare-commit.sh. That'll automatically update the corresponding sip binding and remove the function (since private functions are all stripped from the sip bindings).

It's all nice and automatic now thanks for @3nids. You just need to make sure you either manually run scripts/prepare-commit.sh all the time or set it up as a pre-commit git hook.